core/lib/dal/migrations/20240805144000_tee_proofs_reorg.up.sql

```diff
@@ -0,0 +1,24 @@
+DROP INDEX IF EXISTS idx_tee_proof_generation_details_status_prover_taken_at;
+CREATE TYPE tee_proofs_job_status AS ENUM ('ReadyToBeProven', 'PickedByProver', 'Generated', 'Skipped');
```

Comment:
Is there a strong reason to use ENUM here? I'm afraid of 2 things: inconsistency with zk proofs tables that use `TEXT`; and I'm not sure that using ENUM is flexible enough, e.g. if we want to add a new variant, what will it cost us?
IMHO it's not worth it, strong types don't add much value here.

Reply:
I don't see specific reason why not to use ENUM, just raising it. If others don't have as well then probably we should go with it. But I'd strongly prefer to refactor (not necessarily in one go) all the codebase then. cc @EmilLuta